### PR TITLE
setPedStat for client side peds

### PR DESCRIPTION
**Pull request by ArranTuna**

### CODE REVIEW

File: Client/mods/deathmatch/logic/luadefs/CLuaPedDefs.cpp [hunk 1]
```diff
@@ -2077,6 +2079,57 @@ int CLuaPedDefs::SetPedAimTarget ( lua_State* luaVM )
...
+    float fValue = 0;
+    CScriptArgReader argStream ( luaVM );
+    argStream.ReadUserData ( pPed );
+    argStream.ReadNumber ( usStat );
+    argStream.ReadNumber ( fValue );
+
+    if ( !argStream.HasErrors () )
+    {
+        // Valid ped?
+        if ( pPed )
```
Comment (jushar) [inline Client/mods/deathmatch/logic/luadefs/CLuaPedDefs.cpp]: This line is not necessary as the error case is already handled by `CScriptArgReader::HasErrors`

File: Client/mods/deathmatch/logic/luadefs/CLuaPedDefs.cpp [hunk 2]
```diff
@@ -2077,6 +2079,57 @@ int CLuaPedDefs::SetPedAimTarget ( lua_State* luaVM )
...
+            {
+                pPed->SetStat ( usStat, fValue );
+                lua_pushboolean ( luaVM, true );
+                return 1;
+            }
+            else
+                m_pScriptDebugging->LogCustom ( luaVM, argStream.GetFullErrorMessage ( ) );
+        }
+        else
+            m_pScriptDebugging->LogBadPointer ( luaVM, "ped", 1 );
```
Comment (jushar) [inline Client/mods/deathmatch/logic/luadefs/CLuaPedDefs.cpp]: This can be removed then.

**Review by jushar — COMMENTED** [review]
Looks good all in all.

In addition to the in-code comments, you could avoid the double `m_pScriptDebugging->LogCustom(...)` by something like this:
```
// ...
    if (!argStream.HasErrors())
    {
        pPed->SetStat(usStat, fValue);
        // ...
    }
}

// at the end:
if (argStream.HasErrors())
    m_pScriptDebugging->LogCustom(luaVM, argStream.GetFullErrorMessage());
```

**Comment by ArranTuna** [comment]
Thanks for the review, I've made a new commit. About that if ( pPed ) I assumed that it was needed as many functions in that file have those checks, like getPedStat does. If these checks are useless then can I remove them all in a new patch?

**Comment by ArranTuna** [comment]
I guess something like this needs adding:
MinClientReqCheck ( argStream, MIN_CLIENT_REQ_REMOVEPEDFROMVEHICLE_CLIENTSIDE, "function is being called client side" );

**Comment by ArranTuna** [comment]
I seem to have messed up by using the master branch for a pull request so going to close this and make a new one that uses a branch.

**Pull request closed without merging**